test/replication-luatest/on_replication_split_brain_rollback_event_test.lua

```diff
@@ -0,0 +1,496 @@
+                            local new = stmt.new_tuple
+
+                            t.assert_equals(term, box.info.synchro.queue.term)
+                            t.assert_equals(rid, box.info.id)
+                            if counter == n_stmts then
+                               t.assert_equals(lsn, box.info.lsn)
+                            end
+                            t.assert_equals(sid, box.space.s.id)
+                            t.assert_equals(type, request_type)
+                            t.assert_equals(arg, request_arg)
+                            if request_ops ~= nil then
+                                t.assert_equals(ops, request_ops)
+                            end
+                            if counter == 1 then
+                                t.assert_equals(old, old_tuple)
+                            else
+                                t.assert_equals(old, arg)
+                            end
+                            t.assert_equals(new, new_tuple)
+                            counter = counter + 1
+                        end)
+                    end
+                    t.assert_equals(n_stmts, counter - 1)
+                end
+            end)
+        box.cfg{replication_synchro_quorum = ''}
+        box.space.s:replace{777, 0}
+        box.cfg{replication_synchro_quorum = 3}
+    end)
+end)
+
+-- Test that the event iterator returns correct information for different
+-- transaction statements.
+g.test_different_stmts = function(cg)
+    local stmts = {
+        {'replace', {0}, nil, nil, {0}},
+        {'replace', {777, 1}, nil, {777, 0}, {777, 1}},
+        {'insert', {0}, nil, nil, {0}},
+        {'delete', {0}, nil, nil, nil},
+        {'delete', {777}, nil, {777, 0}, nil},
+        {'update', {777}, {{'=', 2, 1}}, {777, 0}, {777, 1}},
+        {'update', {777}, {{'=', 2, 1}, {'=', 3, 1}}, {777, 0}, {777, 1, 1}},
+        {'upsert', {777}, {{'=', 2, 1}}, {777, 0}, {777, 1}},
+        {'upsert', {0}, {{'=', 2, 1}}, nil, {0}},
+    }
+    for _, stmt in ipairs(stmts) do
+       test_stmt(cg, stmt[1], stmt[2], stmt[3], stmt[4], stmt[5])
+    end
+
+    test_several_stmts(cg)
+end
+
+-- Test that the trigger does not fire during recovery.
+g.test_recovery = function(cg)
+    cause_split_brain(cg.master, cg.replica)
+    cg.master:exec(function()
+        t.helpers.retrying({timeout = 120}, function()
+            t.assert_equals(box.info.synchro.queue.len, 0)
+        end)
+        t.assert(_G.trigger_called)
+    end)
+    cg.master:restart()
+    cg.master:exec(function()
+        t.helpers.retrying({timeout = 120}, function()
+            t.assert_equals(box.info.synchro.queue.len, 0)
+        end)
+        t.assert_not(rawget(_G, 'trigger_called'))
+    end)
+end
+
+g.test_async_committed_async_tx_in_limbo = function(cg)
+    partition_server(cg.replica)
+    cg.master:exec(function()
+        require('fiber').create(function()
+            box.space.s:replace{0}
+        end)
```

Comment:
You must be able to use `submit` with synchronous txns too now. I think it was merged to master already.

Reply:
I need to synchronously commit this transaction.